client/src/utils.ts

```diff
@@ -139,3 +141,81 @@ export function onVisibilityChange(): void {
     });
   }
 }
+
+function getTopicsParams(url = location.href) {
+  // Create a params object
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const recommendedTopics: any = {};
+
+  new URL(url).searchParams.forEach((val, key) => {
+    if (recommendedTopics[key] !== undefined) {
+      if (!Array.isArray(recommendedTopics[key])) {
+        recommendedTopics[key] = [recommendedTopics[key]];
+      }
+      recommendedTopics[key].push(val);
+    } else {
+      recommendedTopics[key] = val;
+    }
+  });
+
+  return recommendedTopics;
+}
+
+export const getRecommendedTopics = (
+  mentorTopics: TopicQuestions[]
+): TopicQuestions => {
+  // 1. get topics from URL
+  const recommendedTopicsURL = getTopicsParams();
+
+  if (recommendedTopicsURL["topicrec"] !== undefined) {
+    //  3. find the topics that match with the mentor's topics
+    const matchedTopics = mentorTopics.filter((topic) => {
+      if (Array.isArray(recommendedTopicsURL["topicrec"])) {
+        return recommendedTopicsURL["topicrec"].some(
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          (recommendedTopics: any) => {
+            return (
+              recommendedTopics.toUpperCase() === topic.topic.toUpperCase()
+            );
+          }
+        );
+      } else {
+        return (
+          recommendedTopicsURL["topicrec"].toUpperCase() ===
+          topic.topic.toUpperCase()
+        );
+      }
+    });
+
+    // 4. get the questions from the matched mentor's topics
+    const recommendedQuestions = matchedTopics.map((topic) => {
+      return topic.questions;
+    });
+
+    // remove the duplicated questions
+    const uniqueQuestions = _.union(
+      recommendedQuestions.reduce((acc, val) => acc.concat(val), [])
+    );
+
+    return { topic: "Recommended", questions: uniqueQuestions };
+  }
+  return { topic: "empty", questions: [] };
+};
+
+export const mergeRecommendedTopicsQuestions = (
+  recommendedTopics: string[],
+  recommendedQuestions: string[]
+): TopicQuestions => {
+  const uniqueQ = Array.from(
+    new Set(recommendedQuestions.map((q) => q.toLowerCase()))
```

Comment:
do these questions get stuck in lower case due to this mapping?

Reply:
I do the mapping before because I need to compare both arrays to remove the duplicated questions. I tried different approaches, but that one is clean using loadash

Reply:
but the questions do get stuck in lower case (except now upper case after your recent changes)
For example, you can run the test "has default 'recommended' topic selected if URL params is avaliable(web - sinlge mentor)" and see the recommended questions coming out as uppercase